ansible/roles/csm.cn.dns-resolver/tasks/main.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Hewlett Packard Enterprise Development LP
+# resolver config tasks
+---
+- block:
+  - name: Slurp in /etc/resolv.conf to get any DHCP added entries
+    slurp:
+      src: "/etc/resolv.conf"
+    register: resolv_conf
+
+  - name: Register /etc/resolv.conf contents
+    set_fact:
+      resolv: "{{ resolv_conf.content | b64decode }}"
+
+  - name: Parse existing options from /etc/resolv.conf
+    set_fact:
+      node_existing_options: "{{ resolv | regex_findall( 'options (\\b(?:[0-9]{1,3}\\.){3}[0-9]{1,3})\\b') }}"
+
+  - name: Slurp in /etc/sysconfig/network/config to get any DHCP added entries
+    slurp:
+      src: "/etc/sysconfig/network/config"
+    register: net_config
+
+  - name: Register /etc/sysconfig/network/config contents
+    set_fact:
+      netconf: "{{ net_config.content | b64decode }}"
+
+  - name: Parse options from config list
+    set_fact:
+      node_netconf_options: "{{ item.split() }}"
+    loop: "{{ netconf | default() | regex_findall( 'NETCONFIG_DNS_RESOLVER_OPTIONS=\"(.*)\"') }}"
+
+  - name: Add existing options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: (node_existing_options is defined) and (item not in node_netconf_options)
+    loop: "{{ node_existing_options }}"
+
+  - name: Add required options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: item not in node_netconf_options
+    loop: "{{ node_required_dns_options }}"
+
+  - name: Add any customer defined options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: item not in node_netconf_options
+    loop: "{{ node_external_dns_options }}"
+
+  - name: Set DNS_RESOLVER_OPTIONS entries
+    when: node_netconf_options is defined
+    lineinfile:
+      dest: /etc/sysconfig/network/config
+      regex: '^NETCONFIG_DNS_RESOLVER_OPTIONS=\"(.*)\"$'
+      line: "NETCONFIG_DNS_RESOLVER_OPTIONS=\"{{ node_netconf_options | join(' ') }}\""
+      backrefs: yes
+
+  - name: Generate updated /etc/resolv.conf by restarting wicked
```

Comment:
Do we _have_ to restart wicked? This will likely kill network activity, which is not ideal.

Reply:
I was thinking of this (and all of the roles borrowed from COS) in the image customization context. It just dawned on me that they can be used for node personalization.
I'll switch it to `shell: netconfig update -f`